qiskit/algorithms/optimizers/umda.py

```diff
@@ -0,0 +1,396 @@
+        """
+        self.__size_gen = value
+
+    @property
+    def max_iter(self) -> int:
+        """Returns the maximum number of iterations"""
+        return self.__max_iter
+
+    @max_iter.setter
+    def max_iter(self, value: int):
+        """
+        Sets the maximum number of iterations of the algorithm.
+
+        Args:
+            value: Maximum number of iterations of the algorithm.
+        """
+        self.__max_iter = value
+
+    @property
+    def alpha(self) -> float:
+        """Returns the alpha parameter value (percentage of population selected to update
+        probabilistic model)"""
+        return self.__alpha
+
+    @alpha.setter
+    def alpha(self, value: float):
+        """
+        Sets the alpha parameter (percentage of individuals selected to update the probabilistic
+        model)
+
+        Args:
+            value: Percentage [0,1] of generation selected to update the probabilistic model.
+        """
+        self.__alpha = value
+
+    @property
+    def n_variables(self) -> int:
+        """Returns the number of variables desired to be optimized"""
+        return self.__n_variables
+
+    @n_variables.setter
+    def n_variables(self, value: int):
+        """
+        Sets the number of variables to be optimized by the UMDA.
+
+        Args:
+            value: Number of variables to be optimized.
+        """
+        self.__n_variables = value
+
+    @property
+    def dead_iter(self) -> int:
+        """Returns the stopping criteria: the number of iterations with no improvement, after
+        which the algorithm converges"""
+        return self.__dead_iter
+
+    @dead_iter.setter
+    def dead_iter(self, value: int):
+        """
+        Sets the stopping criteria: the number of iterations with no improvement after which the
+        algorithm converges and stops.
+
+        Args:
+            value: Number of iterations with no improvement after which the algorithm converges.
+        """
+        self.__dead_iter = value
+
+    @property
+    def best_cost_global(self) -> float:
+        """Returns the best individual cost found until the moment"""
+        return self.__best_cost_global
+
+    @property
+    def best_ind_global(self):
+        """Returns the best individual instance found until the moment"""
+        return self.__best_ind_global
+
+    @property
+    def generation(self):
+        """Returns the actual generation of solutions"""
+        return self.__generation
+
+    @generation.setter
+    def generation(self, value: np.array):
+        """
+        Sets the generation to be evaluated. The user may want to initialize the algorithm from a
+        specific set of solutions.
+
+        Args:
+            value: set of solutions to be set as generation.
+        """
+        self.__generation = value
+
+    @property
+    def history(self) -> list:
+        """Returns the best cost found in each iteration during runtime"""
+        return self.__history
```

Comment:
This makes the optimizer stateful on the last result and this is something we avoid, ie, stateful instances in terms of the solution. They maintain configuration, which can be updated, and then are called upon to do something and all result details are  in the returned object. We used to have stateful instances, like VQE, but a decision was made to move away from that way of behaving and it no longer behaves that way. Maybe create a sub-class of OptimizerResult and store it there if this useful information that users would want.

Reply:
This functionalty has been removed. I do not think is useful for the user. I use the history attribute which now is only local to the minimze function to compute the number of function evaluations to save into the result. Thanks